training/distributed_training/pytorch/model_parallel/GPT2_huggingface_trainer/submit_smp_trainer.ipynb

```diff
@@ -0,0 +1,623 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #30.    mpioptions += "-x FI_EFA_USE_DEVICE_RDMA=1 -x FI_PROVIDER=efa -x RDMAV_FORK_SAFE=1"
</pre><p>p3.16xl does not support EFA. </p><p>Together with the line above, once user is using instance that supports EFA, we should not disable D2D. </p><p>Suggest to move this <code>mpioptions</code> after the config and change it </p><pre class="ql-syntax" spellcheck="false">mpioptions = "-x NCCL_DEBUG=WARN -x SMDEBUG_LOG_LEVEL=ERROR "
if instance_type in ["ml.p3dn.24xlarge", "ml.p4d.24xlarge"]:
   mpioptions += "-x FI_EFA_USE_DEVICE_RDMA=1 -x FI_PROVIDER=efa -x RDMAV_FORK_SAFE=1 "
   if SM_HP_MP_PARAMETERS["pipeline_parallel_degree"] &gt; 1:
       mpioptions += "-x SMP_ENABLE_CROSS_NODE_D2D=1 "

</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3577/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
<p>Ok, sounds good!</p>